packages/react/src/components/List/List.tsx

```diff
@@ -1,24 +1,65 @@
-import type { ComponentPropsWithoutRef } from 'react';
-import React from 'react';
+import type { HTMLAttributes, ReactNode } from 'react';
+import React, { useId } from 'react';
 import cn from 'classnames';
 
+import { Heading, Paragraph } from '../Typography';
+
 import classes from './List.module.css';
 
 export type ListProps = {
-  /** Select which border style between items*/
-  borderStyle?: 'solid' | 'dashed';
-} & ComponentPropsWithoutRef<'ul'>;
+  /**
+   * The type of list to render.
+   * @default ul
+   */
+  as?: 'ul' | 'ol';
+  /** Changes text sizing
+   * @default medium
+   */
+  size?: 'small' | 'medium' | 'large';
+  /**
+   * Heading above the list
+   */
+  heading?: string;
+  /**
+   * Level of the heading
+   * @default 2
+   */
+  headingLevel?: 1 | 2 | 3 | 4 | 5 | 6;
+  children: ReactNode;
+} & HTMLAttributes<HTMLElement>;
 
 export const List = ({
-  borderStyle = 'solid',
   children,
   className,
+  as = 'ul',
+  size = 'medium',
+  heading,
+  headingLevel = 2,
   ...rest
-}: ListProps) => (
-  <ul
-    {...rest}
-    className={cn([classes.list, classes[borderStyle], className])}
-  >
-    {children}
-  </ul>
-);
+}: ListProps) => {
+  const headingId = useId();
+
+  return (
+    <>
+      {heading && (
+        <Heading
+          size={size}
```

Comment:
These look to be the wrong sizes for `Heading`. Looking at the [design in Figma](https://www.figma.com/file/vpM9dqqQPHqU6ogfKp5tlr/Felles-komponenter?type=design&node-id=20798-21781&mode=design&t=fra9w9qyG7ZVeQtl-0) they are `xxsmall`, `xsmall`, `small` respectively.  

Probably just need to make a simple lookup for this translating the component size to heading size